### ISSUE TITLE
“Macro from typed module used in untyped code” when required as language module

### ISSUE THREAD

**Issue by jsmaniac** [racket/typed-racket#181]
When macros from a typed/racket module are imported via a language module, they trigger the message “Macro from typed module used in untyped code”.

```scheme
#lang typed/racket

(module m1 typed/racket
  (provide bar)
  (define-syntax-rule (bar) 42))

(module m2 typed/racket
  (require typed/racket
           (submod ".." m1))
  (provide (all-from-out typed/racket)
           bar))

(module m3 (submod ".." m2)
  (bar)) ;; Type Checker: Macro bar from typed module used in untyped code in: (bar)
```

If the module containing the typed macro is imported directly (and NOT imported via the language module), then they work fine.

```scheme
(module m4 typed/racket
  (require typed/racket)
  (provide (all-from-out typed/racket)))

(module m5 (submod ".." m4)
  (require (submod ".." m1)) ;; m2 works too here.
  (bar)) ;; Works
```
If the macros are defined in the same module as the language module, they fail too:

```scheme
#lang typed/racket

(module m6 typed/racket
  (require typed/racket)
  (define-syntax-rule (bar) 42)
  (provide (all-from-out typed/racket)
           bar))

(module m7 (submod ".." m6)
  (bar)) ;; Type Checker: Macro bar from typed module used in untyped code in: (bar)
```

And if the macros are imported via the language module and are imported separately, they still fail:

```scheme
(module m8 (submod ".." m2)
  (require (submod ".." m1))
  (bar)) ;; Type Checker: Macro bar from typed module used in untyped code in: (bar)
```

**Comment by takikawa** [comment]
I believe this has the same underlying cause as issue #163. I have an idea for a fix which I plan to try to implement soon.

**Comment by takikawa** [comment]
Closed by 241f04bcdb82d46af73890ccca484df3edbf27fa
